overlord/snapstate/backend/link.go

```diff
@@ -68,6 +68,33 @@ type LinkContext struct {
 	// SkipBinaries indicates that we should skip removing snap binaries,
 	// icons and desktop files in UnlinkSnap
 	SkipBinaries bool
+
+	// HasOtherInstances indicates that other instances of the snap are
+	// already installed in the system.
+	HasOtherInstances bool
+}
+
+func createSharedSnapDirForParallelInstance(s snap.PlaceInfo) error {
+	_, key := snap.SplitInstanceName(s.InstanceName())
+
+	if key != "" {
+		err := os.MkdirAll(snap.BaseDir(s.SnapName()), 0755)
+		if err != nil && !os.IsExist(err) {
+			return err
+		}
+	}
+	return nil
+}
+
+func removeSharedSnapDirForParallelInstance(s snap.PlaceInfo) error {
+	_, instanceKey := snap.SplitInstanceName(s.InstanceName())
+
+	if instanceKey != "" {
+		// remove only if not used by other instances of the same snap,
```

Comment:
This comment doesn't quite match the check. It would make more sense in the context of the`cleanupShared...` function below

Reply:
updated the comment